### ISSUE TITLE
Grin binary release fails on CPU without avx2 support

### ISSUE THREAD

**Issue by hashmap** [mimblewimble/grin#2494]
Grin binary release fails silently, when an user runs it one more time it complains about corrupted storage. The real reason is unsupported CPU instruction (SIGILL), most likely avx/avx2 one. 

More info https://github.com/mimblewimble/grin/issues/2334
I create this issue to focus on release process improvement, it seems valuable to have a build with AVX2 support which brings a nice speedup. At the same time it may be confusing for some users if we have 2 packages for each OS.

Here is what we build:

```
$ elfx86exts grin
MODE64 (call)
CMOV (cmova)
SSE1 (movups)
SSE2 (movq)
SSSE3 (pshufb)
AVX (vxorps)
AVX2 (vpcmpeqd)
BMI2 (mulx)
ADX (adcx)
BMI (andn)
CPU Generation: Unknown
```

**Comment by yachoopl** [comment]
I believe AVX is present on core i3 or better (celerons and pentiums don't have it).

**Comment by ignopeverell** [comment]
I'm thinking we're seeing this:

https://github.com/rust-lang/rust/issues/50154

From what I gather, LLVM seems to have some polyfill for architectures that don't support AVX, but their use has seen a regression. It looks like it's been recently fixed in the LLVM repository:

https://bugs.llvm.org/show_bug.cgi?id=37358

The fix is included in LLVM 7.0.1. I'm not 100% sure but perusing the Rust release notes it doesn't seem the fix has been included yet.

**Comment by DinoStray** [comment]
I use docker run grin.
One machine run successfully, another run with the error same as #2334.
The one with no error:
![image](https://user-images.githubusercontent.com/6963679/53414503-e1dd3c80-3a09-11e9-87eb-9ff66e4bd187.png)
The one with error:
![image](https://user-images.githubusercontent.com/6963679/53414570-01746500-3a0a-11e9-9bea-ae747bbf7c81.png)
It seems like that both of them support avx and avx2

**Comment by nikito7** [comment]
Some search

https://gist.github.com/jedisct1/8984243

**Comment by nikito7** [comment]
Fixed?

https://github.com/rust-lang/rust/issues/50154#issuecomment-462948927

**Comment by josef-v** [comment]
We have found out that this might be a problem in croaring crate, which is built natively.
See:
`grep -r native target/release/build/croaring-sys-57163d129afeb5e6/`

`target/release/build/croaring-sys-57163d129afeb5e6/output:running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-Wall" "-Wextra" "-std=c11" "-march=native" "-O3" "-o" "/home/jv/github-repos/grin/target/release/build/croaring-sys-57163d129afeb5e6/out/CRoaring/roaring.o" "-c" "CRoaring/roaring.c"`

**Comment by jaspervdm** [comment]
This should be fixed in v3.1.1 and onwards. If it re-appears, please open a new issue for it and we will investigate
